File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    x = {
        "zero": "0",
        "one": "1",
        "two": "2",
        "three": "3",
        "four": '4',
        "five": "5",
        "six": '6',
        "seven": '7',
        "eight": '8',
        "nine": '9'
    }
    answer = ""
    index = 0
    for i in range(1, len(s) + 1):
        try:
            int(s[index:i])
            answer += s[index:i]
            index = i
        except:
            if s[index:i] in x:
                answer += x[s[index:i]]
                index = i
    return int(answer)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    const words = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    var answer = "";
    let temp = ""
    for (let i = 0; i < s.length; i++) {
        if (isNaN(s[i])) {
            temp += s[i]
            if (words.indexOf(temp) !== -1) {
                answer += words.indexOf(temp).toString()
                temp = ""
            }
        } else {
            answer += s[i]
        }
    }
    return Number(answer);
}